src/opentelemetry/instrumentation/langchain_v2/callback_handler.py

```diff
@@ -278,9 +284,8 @@ def on_llm_end(self,
             ) or response.llm_output.get("model_id")
             if model_name is not None:
                 _set_span_attribute(span, Span_Attributes.GEN_AI_RESPONSE_MODEL, model_name)
-
-                if self.span_mapping[run_id].request_model is None:
-                    _set_span_attribute(span, Span_Attributes.GEN_AI_REQUEST_MODEL, model_name)
+                _set_span_attribute(span, Span_Attributes.GEN_AI_REQUEST_MODEL, model_name)
```

Comment:
sorry I missed this from the last PR. You should also remove this line (request.model should already be set in on_llm_start).